library.rb

```diff
@@ -0,0 +1,150 @@
+require_relative 'rating'
+require_relative 'authors'
+require_relative 'book'
+require_relative 'clients'
+require_relative 'orders'
+require_relative 'file_handler'
+require_relative 'language'
+require 'date'
+require 'time'
+
+class Library < Orders
+  include Language
+  attr_reader :authors_lib, :books_lib, :clients_lib, :orders_lib
+
+  def initialize(authors_lib = Authors.new,
+                 books_lib = Books.new,
+                 clients_lib = Clients.new,
+                 orders_lib = Orders.new)
+    @authors_lib = authors_lib
+    @books_lib = books_lib
+    @clients_lib = clients_lib
+    @orders_lib = orders_lib
+  end
+
+  COMMANDS = ["get_list", "buy_book", "profit", "top_books", "top_authors", "top_clients", "show_books", "show_authors", "show_clients", "show_orders", "exit"]
+
+  def greeting
+    puts phrases_list[:greeting]
+  end
+
+  def commands
+    puts phrases_list[:get_list]
+    COMMANDS.each do |name|
+      p name
+    end
+  end
+
+  def buy_book
+    authors_lib.list_of_authors
+    book_id = books_lib.input_check
+    located_book = books_lib.find_book_by_id(book_id)
+    books_lib.get_book_info(located_book)
+    books_lib.payment_actions
+    client = clients_lib.set_client_info
+    clients_lib.order_creation(client, located_book)
+  end
+
+  def top_books(ntimes, orders_key)
+    books_lib.top_n(ntimes, orders_key, books_lib.books.parse_file, 'id') do |index, hash_top_ids|
+      p "#{index + 1}. #{hash_top_ids['name']}, book id:#{hash_top_ids['id']}"
+    end
+  end
+
+  def top_authors(ntimes, orders_key)
+    authors_lib.top_n(ntimes, orders_key, authors_lib.authors.parse_file, 'book_id') do |index, hash_top_ids|
+      p "#{index + 1}. #{hash_top_ids['first_name']} #{hash_top_ids['last_name']}"
+    end
+  end
+
+  def top_clients(ntimes, orders_key)
+    clients_lib.top_n(ntimes, orders_key, clients_lib.clients.parse_file, 'id') do |index, hash_top_ids|
+      p "#{index + 1}. #{hash_top_ids['first_name']} #{hash_top_ids['last_name']}"
+    end
+  end
+
+  def gain
+    orders_lib.total_profit
+  end
+
+  def set_mode(input)
+    input = input.split('-')
+    if input.size > 1
+      input.delete_at(0)
+      input.join('')
+    else
+      input.join('')
+    end
+  end
+
+  def read_file(file_name)
+    entity = FileHandler.new(file_name).parse_file
+    entity.each { |hash| yield hash }
+    pp entity
+  end
+
+  def show_books
```

Comment:
Better to move it to models [Author, Book...]

Reply:
Done